analyzer/modules/evm.go

```diff
@@ -36,6 +37,63 @@ type EVMBlockTokenData struct {
 	TokenData map[string]*EVMTokenData
 }
 
+type EVMDeterministicError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `EVMDeterministicError{fmt.Errorf("...: %w", err)}` to set up an
+	// instance with `err` in the Unwrap chain.
+	error
+}
+
+func (err EVMDeterministicError) Is(target error) bool {
+	if _, ok := target.(EVMDeterministicError); ok {
+		return true
+	}
+	return false
+}
+
+// TODO: can we move this to oasis-sdk/client-sdk/go/modules/evm?
+const EVMModuleName = "evm"
+
+var (
```

Comment:
Let's add a link to where these come from.

Edit: Searching for the error message `"execution failed"` (with quotes) returns no results in our codebase. I found our EVM SDK module [here](https://github.com/oasisprotocol/oasis-sdk/blob/b37e6da699df331f5a2ac62793f8be099c68469c/runtime-sdk/modules/evm/src/lib.rs#L121-L147), but I'm not clear on what magic removes the `": "` at the end.

Reply:
that horrible magic: https://github.com/oasisprotocol/oasis-sdk/blob/runtime-sdk/v0.2.0/runtime-sdk/modules/evm/src/lib.rs#L147

:arrow_left: that horrible person who wrote it

Reply:
links to sdk added